src/w3c/templates/headers.js

```diff
@@ -89,29 +89,42 @@ const localizationStrings = {
 
 export const l10n = getIntlData(localizationStrings);
 
-function getSpecTitleElem(conf) {
+export function getSpecTitleElem(conf) {
   const specTitleElem =
     document.querySelector("h1#title") || document.createElement("h1");
+
   if (specTitleElem.parentElement) {
     specTitleElem.remove();
+
+    if (specTitleElem.textContent.trim() === "") {
+      pub(
+        "warn",
+        'textContent of h1#title element is "" or whitespace. Falling back to `document.title` or "No Title"'
+      );
+      specTitleElem.textContent = document.title || "No Title";
+    }
+
+    document.title = norm(specTitleElem.textContent);
   } else {
-    specTitleElem.textContent = conf.title;
+    if (document.title) {
+      specTitleElem.textContent = document.title;
+    } else {
+      specTitleElem.textContent = document.title = "No Title";
```

Comment:
Nit: try to avoid these kinds of cascades, as they are unpredictable. For instance, setting document.title has side effects (it strips whitespace). By the end of the chain, you never sure what you are going to get.